add "life of an aws engineer" status quo story (#193)
rust-lang/wg-async-foundations: src/vision/status_quo/aws_engineer/getting_started_with_rust.md

# Status quo of an AWS engineer: Getting started with Rust

For his latest project, Alan is working on a new service, [DistriData]. Like many AWS services, they are trying to move on a tight deadline. 

> The plan is to implement the new service in Rust. The service that they are rewriting was implemented in Java, but it was having difficulty with high tail latencies and other performance hiccups, and they would like to reduce resource usage by adopting Rust.

Author: Good question! I think replacing (and, often rearchitecting) parts of an existing service is much more common.

> The plan is to implement the new service in Rust. The service that they are rewriting was implemented in Java, but it was having difficulty with high tail latencies and other performance hiccups, and they would like to reduce resource usage by adopting Rust.

Reviewer: Makes sense.  Actually as I read the rest of the story, this became clear (since Alan eventually compared the performance of the new rust service to the old java service)

Maybe something like:

```suggestion
The plan is to implement the service in Rust.  This service is actually a rewrite of a service that is implemented in Java, but it is having difficulty with high tail latencies and other performance hiccups, and they would like to reduce resource usage by adopting Rust.
```

> The plan is to implement the new service in Rust. The service that they are rewriting was implemented in Java, but it was having difficulty with high tail latencies and other performance hiccups, and they would like to reduce resource usage by adopting Rust.

Author: I made some tweaks.
